packages/scaffold-config/test/unit/supportFile.spec.ts

```diff
@@ -1,170 +1,223 @@
-    // 'supportFile' configuration option.
-    //
-    // You can read more here:
-    // https://on.cypress.io/configuration
-    // ***********************************************************
-
-    // Import commands.js using ES2015 syntax:
-    import './commands'
-
-    // Alternatively you can use CommonJS syntax:
-    // require('./commands')
-
-    import { mount } from 'cypress/react'
-
-    Cypress.Commands.add('mount', mount)
-
-    // Example use:
-    // cy.mount(<MyComponent />)
-    `)
-  })
-
-  it('handles cypress/react and TS (example with JSX)', () => {
-    const actual = supportFileComponent('ts', findByMountingModule('cypress/react'))
-
-    expect(actual).to.eq(dedent`
-    // ***********************************************************
-    // This example support/component.ts is processed and
-    // loaded automatically before your test files.
-    //
-    // This is a great place to put global configuration and
-    // behavior that modifies Cypress.
-    //
-    // You can change the location of this file or turn off
-    // automatically serving support files with the
-    // 'supportFile' configuration option.
-    //
-    // You can read more here:
-    // https://on.cypress.io/configuration
-    // ***********************************************************
-
-    // Import commands.js using ES2015 syntax:
-    import './commands'
-
-    // Alternatively you can use CommonJS syntax:
-    // require('./commands')
-
-    import { mount } from 'cypress/react'
-
-    // Augment the Cypress namespace to include type definitions for
-    // your custom command.
-    // Alternatively, can be defined in cypress/support/component.d.ts
-    // with a <reference path="./component" /> at the top of your spec.
-    declare global {
-      namespace Cypress {
-        interface Chainable {
-          mount: typeof mount
+  context('react', () => {
+    for (const mountModule of ['cypress/react', 'cypress/react18'] as const) {
+      it(`handles ${mountModule} and JS`, () => {
+        const actual = supportFileComponent('js', mountModule)
+
+        expect(actual).to.eq(dedent`
+          // ***********************************************************
+          // This example support/component.js is processed and
+          // loaded automatically before your test files.
+          //
+          // This is a great place to put global configuration and
+          // behavior that modifies Cypress.
+          //
+          // You can change the location of this file or turn off
+          // automatically serving support files with the
+          // 'supportFile' configuration option.
+          //
+          // You can read more here:
+          // https://on.cypress.io/configuration
+          // ***********************************************************
+
+          // Import commands.js using ES2015 syntax:
+          import './commands'
+
+          // Alternatively you can use CommonJS syntax:
```

Comment:
Unrelated to this PR, but we should probably consider removing this line, eventually. I am not even sure if Vite 3 works with `require` out of the box (maybe it does, I haven't tried). We don't have any tests for it, though, so we probably don't want to recommend something unless we know it works 100% of the time.

Reply:
Created an issue: https://github.com/cypress-io/cypress/issues/23287